### ISSUE TITLE
Possible to use with other Chromium webviews?

### ISSUE THREAD

**Issue by Darthagnon** [i5ik/graderjs#19]
I noticed in this projects' readme and the readmes of related projects that Google Chrome is a dependency.

Is it possible to configure GraderJS programs to run with alternative Chrome versions, e.g.:
- [Edgium Webview](https://developer.microsoft.com/en-us/microsoft-edge/webview2/)
- [Ungoogled Chromium](https://chromium.woolyss.com/) (or [here](https://ungoogled-software.github.io/ungoogled-chromium-binaries/releases/windows/64bit/))
- [Chrome Webdriver](https://chromedriver.chromium.org/downloads)

I'd prefer to avoid using vanilla Google Chrome due to privacy concerns. Microsoft (Edgium) isn't much better, but they make my operating system, too, so it seems the lesser evil.

Thank you for making this; I've often thought "Why do web service programs have to bundle Electron/CEF when they could defer to a preinstalled browser's webview?"

**Comment by i5ik** [comment]
It's a good question. Thank you @Darthagnon for this! I think it should be possible. Everything that's Chrome (chromium, etc, even webviews) should be able to be connected to via DevTools. The question, I suppose, is how to expose that?

With a browser (Chrome, Chromium, "Edgium"--heh) we just need to pass a command-line argument `--remote-debugging-port=${some_port}`; for WebViews I'm not so sure. For ChromeDriver--that's not a browser nor a WebView, right? It's more like an API to instrument chrome, similar to DevTools or puppeteer, as far as I know--I don't think idea to incorporate it applies, as you still need something, a renderer (like a browser or webview) to connect the driver to. 

I'm happy for you to look into the feasibility of adding support for WebViews, and if you choose to go ahead with that and post your progress back here I'll re-open this issue to maybe act as a record of that progress. If you manage to get webviews working (which I guess would involve automatically detecting a WebView on a client's system, incorporating the WebView in the UI using NodeJS, connecting to it, and using it to render the app content) that would be wonderful. It may be easier than it sounds, but it also may be a lot of work!

Right now I think the browser requirement is a reasonable solution that balances being widely accessible, easy to locate on the system, easy to install if not located, and easy to connect to and use as a renderer. But I don't know about alternatives to browsers, they may be just as easy, I don't have that knowledge of working with them. So it might be a good strategy to look into that as you suggest. 

Another alternative if we wanted to diversity our dependence away from just Chrome/Chromium would be to figure out how to utilize other browsers installed on the system, such as Firefox. This should be possible as [Firefox implements at least some of Chrome DevTools Protocol / Remote Debugging Protocol](https://firefox-source-docs.mozilla.org/devtools/backend/protocol.html#remote-debugging-protocol). Also it seems like a generalization of the DevTools protocol is [on track to maybe become](https://github.com/w3c/webdriver) a W3C standard (so more browsers may implement it). 

To sum up I'm very interested in knowing if this was feasible and how it would work. It could be a fascinating, and very useful, more lightweight alternative to a browser requirement. And I heartily endorse us looking into this--if you want to take the lead on that--and if you got somewhere with it, I would definitely consider including it, because I think it's a great feature if it works.
